lib/adapters/http.js

```diff
@@ -145,17 +172,8 @@ module.exports = function httpAdapter(config) {
     }
 
     if (proxy) {
-      options.hostname = proxy.host;
-      options.host = proxy.host;
       options.headers.host = parsed.hostname + (parsed.port ? ':' + parsed.port : '');
```

Comment:
Very minor thing but it might be neater and remove some of the duplication if you stuffed the result the host string building into it's own variable.
`
var host = parsed.hostname + parsed.port ? ':' + parsed.port : ''';
`
Which results in
`
 options.headers.host = host;
 options = setProxy(options, proxy, protocol + '//' + host + options.path);
`